pkg/lib/authn/authenticator/password/expiry.go

```diff
@@ -0,0 +1,25 @@
+package password
+
+import (
+	"github.com/authgear/authgear-server/pkg/lib/authn/authenticator"
+	"github.com/authgear/authgear-server/pkg/lib/config"
+
+	"github.com/authgear/authgear-server/pkg/util/clock"
+)
+
+type Expiry struct {
+	ForceChangeEnabled         bool
+	ForceChangeSinceLastUpdate config.DurationString
+	Clock                      clock.Clock
+}
+
+func (pe *Expiry) Validate(authenticator *authenticator.Password, password string) error {
+	if IsSamePassword(authenticator.PasswordHash, password) {
```

Comment:
No need to verify the password once more. The password should have been verified before this method is called.

Comparing password is an expensive operation, we do not want to spend time doing it twice.